hydromt/models/_v1/model_root.py

```diff
@@ -0,0 +1,115 @@
+"""the new model root class."""
+
+from logging import Logger, getLogger
+from os import PathLike
+from os.path import isdir
+from pathlib import Path
+
+from hydromt.io.path import make_path_abs_and_cross_platform
+from hydromt.typing import ModeLike, ModelMode
+
+logger = getLogger(__name__)
+
+
+class ModelRoot:
+    """A class to handle model roots in a cross platform manner."""
+
+    def __init__(
+        self,
+        path: PathLike,
+        mode: ModeLike = "w",
+        logger: Logger = logger,
+    ):
+        self.logger: Logger = logger
+        self.set_root(path, mode)
```

Comment:
Options to use python-style setters and getters using `@property`. Setting in python generally works with assignments like `model_root.mode = 'w'`